Add MAX_SUBSCRIBE_ID to limit number of subscriptions (#491)
moq-wg/moq-transport: draft-ietf-moq-transport.md

given URI by setting up a QUIC connection to the host and port
identified by the `authority` section of the URI.  The `path-abempty`
and `query` portions of the URI are communicated to the server using the

> PATH parameter ({{path}}) which is sent in the CLIENT_message at the

Author: ```suggestion
PATH parameter ({{path}}) which is sent in the CLIENT_SETUP message at the
```

* Duplicate Track Alias: The endpoint attempted to use a Track Alias
  that was already in use.

* Too Many Subscribes: The session was closed because the subscriber used

>   a Subscribe ID larger than the current Maximum Subscribe ID.

Author: ```suggestion
  a Subscribe ID equal or larger than the current Maximum Subscribe ID.
```

>   a Subscribe ID equal or larger than the current Maximum Subscribe ID.

Reviewer: Is it expected that a publisher MUST close an entire session under these circumstances? Or could it just not fulfill the subscription request? I guess maybe we don't have another way to signal the latter.

Reviewer: My concern would be that the potential impact on aggregating subscribers (relays) could be substantial if they erroneously open too many subscriptions (or are somehow coerced into doing so).

Reviewer: That said, having a limit here is better than no limit, so I'd be fine revisiting this concern at a later date.

Author: Aggregating subscribers exist in HTTP proxies as well, and those deal with stream limits on both upstream and downstream fine, so I'd suggest we revisit if it turns out to be a problem people experience.

URI; if `query` is present, the client MUST concatenate `?`, followed by
the `query` portion of the URI to the parameter.

> #### MAX_SUBSCRIBE_ID {#max-subscribe-id}

Reviewer 2: This is unnecessary, since you can just send a MAX_SUBSCRIBE_ID immediately after the SETUP message.

Author: Yes.  I added it because we have the equivalent in QUIC.

* Final Object: The largest Object ID sent by the publisher in an OBJECT
message in the `Final Group` for this track.

## MAX_SUBSCRIBE_ID {#message-max-subscribe-id}

A publisher sends a MAX_SUBSCRIBE_ID message to increase the number of
subscriptions a subscriber can create within a session.

The Maximum Subscribe Id MUST only increase within a session, and
receipt of a MAX_SUBSCRIBE_ID message with an equal or smaller Subscribe ID
value is a 'Protocol Violation'.

~~~
MAX_SUBSCRIBE_ID
{
  Subscribe ID (i),
}
~~~
{: #moq-transport-max-subscribe-id format title="MOQT MAX_SUBSCRIBE_ID Message"}

> * Subscribe ID: The maximum valid Subscribe ID for the session. If a Subscribe ID

Author: ```suggestion
* Subscribe ID: The new Maximums Subscribe ID for the session. If a Subscribe ID
```

* Final Object: The largest Object ID sent by the publisher in an OBJECT
message in the `Final Group` for this track.

## MAX_SUBSCRIBE_ID {#message-max-subscribe-id}

A publisher sends a MAX_SUBSCRIBE_ID message to increase the number of
subscriptions a subscriber can create within a session.

The Maximum Subscribe Id MUST only increase within a session, and
receipt of a MAX_SUBSCRIBE_ID message with an equal or smaller Subscribe ID

> value is a 'Protocol Violation'.

Reviewer 3: If a subscriber received  MAX_SUBSCRIBE_ID with value lesser than previous MAX_SUBSCRIBE_ID sent by the publisher, are we saying here that the subscriber needs to close the session ? I am not sure if that is feasible to do or can be enforced.
